### DISCUSSION TITLE
did hugo v0.100+ break geekdoc

### DISCUSSION THREAD

**Discussion by wolfspyre** [thegeeklab/hugo-geekdoc#449] [Q&A]
[The release notes for hugo v0.100](https://github.com/gohugoio/hugo/releases/tag/v0.100.0) point out a change to `$page.RenderString` facilitating shortcode implementation, and after upgrading hugo locally:

```
Error: Error building site: "/Users/me/site/docs/content/Snippits/Geekdocs/Shortcodes/katex.md:9:1":  
failed to render shortcode "columns": failed to process shortcode: "/Users/me/site//docs/themes/hugo-geekdoc/layouts/shortcodes/columns.html:4:14":  
execute of template failed: template: shortcodes/columns.html:4:14:  
executing "shortcodes/columns.html" at <$.Page.RenderString>:  
error calling RenderString: "/Users/me/site/docs/content/Snippits/Geekdocs/Shortcodes/katex.md:4:11":  
unrecognized character in shortcode action: U+005B '['.  
Note: Parameters with non-alphanumeric args must be quoted
Built in 1128 ms
```

The content of that file:

```markdown
---
title: KaTeX
---

[KaTeX](https://katex.org/) shortcode let you render math typesetting in markdown document.

## Example

{{< columns >}}

# These hashes added to have Github render this properly
# ```latex
# {{</* katex [display] [class="text-center"] */>}}
# f(x) = \int_{-\infty}^\infty\hat f(\xi)\,e^{2 \pi i \xi x}\,d\xi
# {{</* /katex */>}}
# ```
<--->

<!-- spellchecker-disable -->
<!-- prettier-ignore -->
{{< katex display >}}
f(x) = \int_{-\infty}^\infty\hat f(\xi)\,e^{2 \pi i \xi x}\,d\xi
{{< /katex >}}

<!-- spellchecker-enable -->

{{< /columns >}}

KaTeX can also be used inline, for example {{< katex >}}\pi(x){{< /katex >}} or used with the `display` parameter (e.g. `display: block`):

<!-- spellchecker-disable -->
<!-- prettier-ignore -->
{{< katex display >}}
f(x) = \int_{-\infty}^\infty\hat f(\xi)\,e^{2 \pi i \xi x}\,d\xi
{{< /katex >}}

<!-- spellchecker-enable -->

Text continues here.
```

I suspect the two are related? but ¯\\\_(ツ)\_/¯

Should I raise an issue about this? 
or am I doin-it-wrong somehow?

**Comment by xoxys** [comment]
Right now this looks more like a Hugo issue, no idea what's the correct way to escape the shortcode syntax to e.g. use it for documentation in a code block... Let's wait for https://github.com/gohugoio/hugo/issues/10058

**Reply by xoxys** [reply]
As a workaround, quoted shortcode blocks can be quoted twice e.g. `{{</*/* katex [display] [class="text-center"] */*/>}}` instead of `{{</* katex [display] [class="text-center"] */>}}`

**Comment by jamesbraza** [comment]
I am hitting this today with Hugo version `0.114.0` with `hi.md`

```md
{{< toc >}}

## Test

Test
```

Leads to:

```
 assemble: [1;36m"/path/to/repo/content/en/2022/_index.md:1:1"[0m: unrecognized character in shortcode action: U+0028 '('. Note: Parameters with non-alphanumeric args must be quoted  assemble: [1;36m"/path/to/repo/content/en/2022/hi.md:1:1"[0m: unmarshal failed: invalid character '{' looking for beginning of object key string 
```

Moving to `{{</*/* toc */*/>}}` workaround posed above did not work :/

**Reply by jamesbraza** [reply]
Backdating to Hugo version `0.99.1` with Go version `1.20.5`, I can't build due to `template for shortcode "button" not found`:

```shell
> git clone git@github.com:gohugoio/hugo.git && cd hugo
...
> git checkout tags/v0.99.1 -b tags-v0.99.1
...
> go install
...
> hugo server -c path/to/repo -D
Start building sites …
hugo v0.99.1-d524067382e60ce2a2248c3133a1b3af206b6ef1 darwin/arm64 BuildDate=2022-05-18T11:18:14Z
Error: Error building site: "/path/to/repo/content/en/_index.md:10:1": failed to extract shortcode: template for shortcode "button" not found
Built in 886 ms
```

@xoxys any tips on how to get out of this bind?  I would like to use both `button` and `toc`

---

UPDATE: the issue here was not passing the themes directory argument to `hugo server`.  I decided to just add `hugo` to path, and invoke `hugo` from the actual project directory (and not from `git clone` location for Hugo)

**Reply by xoxys** [reply]
Not able to reproduce it. Geekcdocs documentation is using buttons and tocs and builds fine  https://github.com/thegeeklab/hugo-geekdoc/blob/main/.drone.yml#L210

**Reply by jamesbraza** [reply]
Yeah actually, I was just testing `0.105.0` from https://github.com/gohugoio/hugo/releases/download/v0.105.0/hugo_0.105.0_darwin-universal.tar.gz.  Unfortunately it's not yet working there either:

```shell
> export PATH=/path/to/Downloads/hugo_0.105.0_darwin-universal:$PATH
> hugo version
hugo v0.105.0-0e3b42b4a9bdeb4d866210819fc6ddcf51582ffa darwin/arm64 BuildDate=2022-10-28T12:29:05Z
> hugo server -D
Start building sites …
hugo v0.105.0-0e3b42b4a9bdeb4d866210819fc6ddcf51582ffa darwin/arm64 BuildDate=2022-10-28T12:29:05Z
Error: Error building site: "/path/to/repo/content/en/2022/file.md:17:1": failed to render shortcode "include": failed to process shortcode: "/path/to/repo/themes/hugo-geekdoc/layouts/shortcodes/include.html:10:27": execute of template failed: template: shortcodes/include.html:10:27: executing "shortcodes/include.html" at <readFile>: error calling readFile: file does not exist
Built in 98 ms
```

The relevant message being:

```shell
executing "shortcodes/include.html" at <readFile>: error calling readFile: file does not exist
```

---

UPDATE: the issue here was I actually had a broken `include` path that Hugo 0.114.0 seems to have been masking, but Hugo 0.105.0 was not.

**Reply by xoxys** [reply]
If you provide a minimal reproduction repo I can take a look.

**Reply by jamesbraza** [reply]
I figured out both issues, `toc` shortcode works fine with Hugo 0.105.0.

I edited the above posts with UPDATE to detail where I was going wrong
